tests/local/shared-everything-threads/global.wast

```diff
@@ -12,26 +48,363 @@
+  )
+  "invalid type")
+
+(assert_invalid
+  (module
+    (global $a (shared mut f64) (f64.const 0))
+    (func (global.atomic.set acq_rel $a (f64.const 0)))
+  )
+  "invalid type")
+
+(assert_invalid
+  (module
+    (global $a (shared f64) (f64.const 0))
+    (func (result f64) (global.atomic.get seq_cst $a))
+  )
+  "invalid type")
+
+(assert_invalid
+  (module
+    (global $a (shared f64) (f64.const 0))
+    (func (result f64) (global.atomic.get acq_rel $a))
+  )
+  "invalid type")
+
+(assert_invalid
+  (module
+    (global $a (shared mut v128) (v128.const i64x2 0 0))
+    (func (result v128) (global.atomic.get seq_cst $a))
+  )
+  "invalid type")
+
+(assert_invalid
+  (module
+    (global $a (shared mut v128) (v128.const i64x2 0 0))
+    (func (global.atomic.set seq_cst $a (v128.const i64x2 0 0)))
+  )
+  "invalid type")
+
+(assert_invalid
+  (module
+    (global $a (shared mut v128) (v128.const i64x2 0 0))
+    (func (result v128) (global.atomic.get acq_rel $a))
+  )
+  "invalid type")
+
+(assert_invalid
+  (module
+    (global $a (shared mut v128) (v128.const i64x2 0 0))
+    (func (global.atomic.set acq_rel $a (v128.const i64x2 0 0)))
+  )
+  "invalid type")
+
+(assert_invalid
+  (module
+    (global $a (shared v128) (v128.const i64x2 0 0))
+    (func (result v128) (global.atomic.get seq_cst $a))
+  )
+  "invalid type")
+
+(assert_invalid
+  (module
+    (global $a (shared v128) (v128.const i64x2 0 0))
+    (func (result v128) (global.atomic.get acq_rel $a))
+  )
+  "invalid type")
+
+;; Check global.atomic.rmw.*.
+(module
+  (global $b (import "spectest" "global_i32") (shared mut i32))
+  (global $a (shared mut i32) (i32.const 0))
+  (func (export "rmw-add-i32-seq_cst-$a") (param $x i32) (param $y i32) (result i32)
+    local.get $x
+    local.get $y
+    global.atomic.rmw.add seq_cst $a)
+  (func (export "rmw-add-i32-seq_cst-$b") (param $x i32) (param $y i32) (result i32)
+    local.get $x
+    local.get $y
+    global.atomic.rmw.add seq_cst $b)
+  (func (export "rmw-add-i32-acq_rel-$a") (param $x i32) (param $y i32) (result i32)
+    local.get $x
+    local.get $y
+    global.atomic.rmw.add acq_rel $a)
+  (func (export "rmw-add-i32-acq_rel-$b") (param $x i32) (param $y i32) (result i32)
+    local.get $x
+    local.get $y
+    global.atomic.rmw.add acq_rel $b)
+  (func (export "rmw-sub-i32-seq_cst-$a") (param $x i32) (param $y i32) (result i32)
+    local.get $x
+    local.get $y
+    global.atomic.rmw.sub seq_cst $a)
+  (func (export "rmw-sub-i32-seq_cst-$b") (param $x i32) (param $y i32) (result i32)
+    local.get $x
+    local.get $y
+    global.atomic.rmw.sub seq_cst $b)
+  (func (export "rmw-sub-i32-acq_rel-$a") (param $x i32) (param $y i32) (result i32)
+    local.get $x
+    local.get $y
+    global.atomic.rmw.sub acq_rel $a)
+  (func (export "rmw-sub-i32-acq_rel-$b") (param $x i32) (param $y i32) (result i32)
+    local.get $x
+    local.get $y
+    global.atomic.rmw.sub acq_rel $b)
```

Comment:
Shouldn't this, and others, only take one operand on the stack? It's the global -= one stack value, not two stack values? (I think cmpxchg is the only one taking two operands)